### PR TITLE
JDK-8306578: Report error if no label given in @see and {@link} when no default is available

### PR DESCRIPTION
**Pull request by hns**
Please review a trivial change to generate an error instead of a warning when no label is provided with a reference to a user-provided anchor.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8306578](https://bugs.openjdk.org/browse/JDK-8306578): Report error if no label given in @see and {@link} when no default is available

### Reviewers
 * [Jonathan Gibbons](https://openjdk.org/census#jjg) (@jonathan-gibbons - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/13720/head:pull/13720` \
`$ git checkout pull/13720`

Update a local copy of the PR: \
`$ git checkout pull/13720` \
`$ git pull https://git.openjdk.org/jdk.git pull/13720/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 13720`

View PR using the GUI difftool: \
`$ git pr show -t 13720`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/13720.diff">https://git.openjdk.org/jdk/pull/13720.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/13720#issuecomment-1527645306)

### CODE REVIEW

**Comment by jonathan-gibbons** [comment]
Although not a conflict, see PR #13811 for a related change in this area.

File: src/jdk.javadoc/share/classes/jdk/javadoc/internal/doclets/formats/html/TagletWriterImpl.java [hunk 1]
```diff
@@ -500,7 +500,7 @@ private Content linkSeeReferenceOutput(Element holder,
             if (labelContent.isEmpty()) {
                 // A non-empty label is required for fragment links as the
                 // reference target does not provide a useful default label.
-                reportWarning.accept("doclet.link.see.no_label", null);
+                htmlWriter.messages.error(ch.getDocTreePath(refTree), "doclet.link.see.no_label");
```
[the remark this answers is not in the document]
Comment (jonathan-gibbons) [reply src/jdk.javadoc/share/classes/jdk/javadoc/internal/doclets/formats/html/TagletWriterImpl.java:497]: Filed https://bugs.openjdk.org/browse/JDK-8309158

[the remark this answers is not in the document]
Comment (hns) [reply src/jdk.javadoc/share/classes/jdk/javadoc/internal/doclets/formats/html/TagletWriterImpl.java:497]: As far as I understand, the considerations about `reportWarning` do not affect this change. Is there anything that should be changed in this PR?

[the remark this answers is not in the document]
Comment (jonathan-gibbons) [reply src/jdk.javadoc/share/classes/jdk/javadoc/internal/doclets/formats/html/TagletWriterImpl.java:497]: No, I do not see what we can/should change in this PR at this time.

I do think we should subsequently review how we generate diagnostics in this method.

**Review by jonathan-gibbons — APPROVED** [review]

**Pull request closed without merging**